### PR TITLE
Update member API doc (fixes #[8087])

### LINKED ISSUE #8087: Improve API Documentation   (apidocs)
_EDIT by Alys 2018-06-30:_
 
The only remaining action for this PR is to document the member route. It's been started (probably mostly done) by @madpink in https://github.com/HabitRPG/habitica/pull/9760 so that needs to be taken over by someone and finished (see the staff comments in that PR).

----

The docs for the API routes are, for the most part, very sparse. Often they don't provide enough information for actually using them.

There are a bunch of parameters that we can use with [API Doc](http://apidocjs.com/#params).

A good example of how each route should look is the [invite to group route](https://habitica.com/apidoc/#api-Group-InviteToGroup). Here it is in the [source code](https://github.com/HabitRPG/habitrpg/blob/ee15e29ba4d34812c4fd4e5836dc310d713a7db9/website/server/controllers/api-v3/groups.js#L643-L683).

When opening a PR, you should do it a little at a time. NOT ALL AT ONCE. For instance, one PR for each section.
- [x] Challenge -  @MathWhiz -- _(a mostly complete solution by JTorr is at https://github.com/HabitRPG/habitica/pull/8112 and if MathWhiz uses it in a PR that is merged, JTorr will get partial credit towards a tier)_  https://github.com/HabitRPG/habitica/pull/8626
- [x] Chat - @MathWhiz https://github.com/HabitRPG/habitica/pull/8277
- [x] Content (just document the top level parts of the object) - @MathWhiz #8098
- [x] Coupon - @MathWhiz https://github.com/HabitRPG/habitica/pull/8109
- [x] Data Export - @MathWhiz https://github.com/HabitRPG/habitica/pull/8120
- [x] Development - @MathWhiz https://github.com/HabitRPG/habitica/pull/8480
- [x] Group - @MathWhiz https://github.com/HabitRPG/habitica/pull/8522
  - [x] Invite to Group - @JTorr #8091
- [x] Hall - @Althenis #8536
- [ ] Member - continue from https://github.com/HabitRPG/habitica/pull/9760
- [x] Meta - @MathWhiz https://github.com/HabitRPG/habitica/pull/8167
- [x] Status - @MathWhiz #8165 
- [x] Tag - @madpink #8400 
- [x] Task - @madpink https://github.com/HabitRPG/habitica/pull/8447
- [x] Unsubscribe - @MathWhiz #8187 
- [x] User - a maximum of 10 routes per PR:
   - [x] [GET `/api/v3/user`](https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/user.js#L26) – [POST `/api/v3/user/buy/:key`](https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/user.js#L560) - @madpink #8476
   - [x] [POST `/api/v3/user/buy-gear/:key`](https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/user.js#L580) – [POST `/api/v3/user/change-class`](https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/user.js#L785) @madpink https://github.com/HabitRPG/habitica/pull/8602
   - [x] [POST `/api/v3/user/disable-classes`](https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/user.js#L810) – [POST `/api/v3/user/unlock`](https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/user.js#L1007) @madpink https://github.com/HabitRPG/habitica/pull/8720
   - [x] [POST `/api/v3/user/revive`](https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/user.js#L1031) – [POST `/api/v3/user/custom-day-start`](https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/user.js#L1267) @madpink https://github.com/HabitRPG/habitica/pull/8792

### PR DESCRIPTION
**Pull request by JacobFrericks**
Fixes #8087
Closes #9760 

### Changes
Adding some more API documentation to the member API

----
UUID:  6c4ef2c1-2ae1-41a4-85cc-35228c926366

### CODE REVIEW

**Comment by JacobFrericks** [comment]
Note: A lot of code was taken from https://github.com/HabitRPG/habitica/pull/9760 as most of the work was done already. I took the comment into account and agreed that we should keep the `@apiuses`.

File: website/server/controllers/api-v3/members.js [hunk 1]
```diff
@@ -519,9 +647,9 @@ api.sendPrivateMessage = {
  * @apiName TransferGems
  * @apiGroup Member
  *
- * @apiParam (Body) {String} message The message
```
Comment (paglias) [inline website/server/controllers/api-v3/members.js:522]: why were all the (Body), (Path) and (Query) removed?

Comment (JacobFrericks) [reply website/server/controllers/api-v3/members.js:522]: @paglias the example API didn't have these, so I assumed they were taken out for a reason. 
https://github.com/HabitRPG/habitica/blob/ee15e29ba4d34812c4fd4e5836dc310d713a7db9/website/server/controllers/api-v3/groups.js#L643-L683

I don't mind putting them back in if that's what we want to do, though.

Comment (SabreCat) [reply website/server/controllers/api-v3/members.js:522]: We have them throughout the rest of the documentation (see e.g. all the `@apiParam` entries in `website/server/controllers/api-v3/tasks.js`), so I'd say it's better to have them.

Comment (JacobFrericks) [reply website/server/controllers/api-v3/members.js:522]: @paglias @SabreCat Fixed!

**Comment by SabreCat** [comment]
Thanks for wrapping this up, @JacobFrericks! Noted as credit toward your second contributor tier!

**Pull request merged**